### ISSUE TITLE
Search for accented characters (diacritics)

### ISSUE THREAD

**Issue by envas** [OFFLINE-GmbH/oc-site-search-plugin#147]
I noticed that the search criteria must be defined with diacritics for accented languages. Do you know any trick for finding the occurrence of `préposition` with the search criteria `preposition`?

**Comment by tobias-kuendig** [comment]
You could try somethink like this or fall back to a proper fulltext search. Both options can have considerable downsides depending on the size of your database:

https://stackoverflow.com/a/8649500

**Comment by envas** [comment]
Your hint refers to MySQL. We are using Postgres with the unaccent extension, where the search condition can be easily defined as

```
select *  from tableX where unaccent(column) like unaccent('%my_query_string%')
```

I need help with how to modify the SQL query before sending it to the database.

**Comment by tobias-kuendig** [comment]
You'd have to disable all default providers and provide your own versions. Most providers use a method that returns the results, so you'd only have to override this method in your custom provider. For example: https://github.com/OFFLINE-GmbH/oc-site-search-plugin/blob/develop/classes/providers/RainlabPagesResultsProvider.php#L65

**Comment by envas** [comment]
OK, I will try it, I need only Tailor search

**Comment by tobias-kuendig** [comment]
In this case you can copy the class 1:1 and just modify the queries. Also make sure to disable the Tailor module in the backend settings and `return true` from the `isEnabled()` method in your own implementation:

https://github.com/OFFLINE-GmbH/oc-site-search-plugin/blob/develop/classes/providers/TailorResultsProvider.php

**Comment by envas** [comment]
I have no Tailor provider in the backend settings. That's all I see there, only CMS pages. However, Tailor search works (probably default enabled)

![CleanShot 2024-07-23 at 15 36 19@2x](https://github.com/user-attachments/assets/55619fcb-68de-4e45-9f24-76aebc9efeff)

**Comment by tobias-kuendig** [comment]
That was an oversight, it is now available in v1.7.20

**Comment by envas** [comment]
Well, quick and dirty (currently Postgres only) works! Replaced line [52](https://github.com/OFFLINE-GmbH/oc-site-search-plugin/blob/5caf2e209885071cfcb652a3624e6e1107ce449c/classes/providers/TailorResultsProvider.php#L52l) in Tailor provider with following code. Additionally, I added the lower() expression to compare case-insensitively. 

```
// works only on PostgreSQL !
$cond = "lower(unaccent(" . $field . ")) like lower(unaccent('%" . $this->query . "%'))";
$q->orWhereRaw($cond);
```

I appreciate your support. Best Regards from Emmenbrücke; we're almost neighbours.
